bundle/src/main/java/com/adobe/acs/commons/properties/impl/PropertyAggregatorServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+                addInheritedPagePropertiesToMap(map, currentPage);
+                addAdditionalDataToMap(map, currentPage);
+            }
+        }
+
+        return map;
+    }
+
+    @Override
+    public Map<String, Object> getProperties(final Page page) {
+        return getProperties(page.getContentResource());
+    }
+
+    /**
+     * Iterate over the properties of a bean and add the property values (retrieved by invoking the
+     * method) to a map with the 'prefix.property' as the key.
+     * @param targetPropertyMap the map that should be updated with the properties and their values
+     * @param key the current nested name for the property (each call will append the current property name to the parent's separated by a period)
+     * @param beanValue the current property's value
+     */
+    private void addBeanToMap(final Map<String,Object> targetPropertyMap, final String key, final Object beanValue) {
+        if (targetPropertyMap == null || StringUtils.isBlank(key) || beanValue == null) {
+            return;
+        }
+        try {
+            Class<?> beanClass = Class.forName(beanValue.getClass().getName());
+            BeanInfo beanInfo = Introspector.getBeanInfo(beanClass);
+            PropertyDescriptor[] propertyDescriptors = beanInfo.getPropertyDescriptors();
+
+            for (PropertyDescriptor propertyDescriptor : propertyDescriptors) {
+                if (!"class".equalsIgnoreCase(propertyDescriptor.getName())) {
+                    String newKey = key + "." + propertyDescriptor.getName();
+                    Object propertyValue = propertyDescriptor.getReadMethod().invoke(beanValue);
+                    if (propertyValue != null) {
+                        targetPropertyMap.put(newKey, propertyValue);
+                    }
+                }
+            }
+
+        } catch (ClassNotFoundException e) {
+            log.error("Error mapping to class " + beanValue.getClass().getName(), e);
+        } catch (IntrospectionException e) {
+            log.error("IntrospectionException", e);
+        } catch (IllegalAccessException e) {
+            log.error("IllegalAccessException", e);
+        } catch (InvocationTargetException e) {
+            log.error("InvocationTargetException", e);
+        }
+    }
+
+    /**
+     * Add the properties of a page to the given map.  Excluded properties are found in the configuration for this service.
+     * @param targetPropertyMap the map that should be updated with the properties and their values
+     * @param page the page containing properties
+     */
+    private void addPagePropertiesToMap(final Map<String,Object> targetPropertyMap, final Page page, final String prefix) {
+        ValueMap pageProperties = page.getProperties();
+        Set<Map.Entry<String, Object>> entries = pageProperties.entrySet();
+        for (Map.Entry<String, Object> entry : entries) {
+            if (notInMap(targetPropertyMap, entry.getKey())
```

Comment:
What is the purpose of the `notInMap` check?

Reply:
This is to check if a property with the same local name exists already overridden in the map of properties. This is so we don't get both `inherited_page_properties.property` and `page_properties.property`.